src/java.base/macosx/classes/sun/nio/fs/BsdFileAttributeViews.java

```diff
@@ -50,94 +50,63 @@ private static void setTimes(UnixPath path, FileTime lastModifiedTime,
         // permission check
         path.checkWrite();
 
-        boolean useLutimes = false;
-        try {
-            useLutimes = !followLinks &&
-                UnixFileAttributes.get(path, false).isSymbolicLink();
-        } catch (UnixException x) {
-            x.rethrowAsIOException(path);
-        }
-
-        int fd = -1;
-        if (!useLutimes) {
-            try {
-                fd = path.openForAttributeAccess(followLinks);
-            } catch (UnixException x) {
-                x.rethrowAsIOException(path);
-            }
-        }
-
-        try {
-            // not all volumes support setattrlist(2), so set the last
-            // modified and last access times using futimens(2)/lutimes(3)
-            if (lastModifiedTime != null || lastAccessTime != null) {
-                // if not changing both attributes then need existing attributes
-                if (lastModifiedTime == null || lastAccessTime == null) {
-                    try {
-                        UnixFileAttributes attrs = UnixFileAttributes.get(fd);
-                        if (lastModifiedTime == null)
-                            lastModifiedTime = attrs.lastModifiedTime();
-                        if (lastAccessTime == null)
-                            lastAccessTime = attrs.lastAccessTime();
-                    } catch (UnixException x) {
-                        x.rethrowAsIOException(path);
-                    }
-                }
-
-                // update times
-                TimeUnit timeUnit = useLutimes ?
-                    TimeUnit.MICROSECONDS : TimeUnit.NANOSECONDS;
-                long modValue = lastModifiedTime.to(timeUnit);
-                long accessValue= lastAccessTime.to(timeUnit);
-
-                boolean retry = false;
+        // not all volumes support setattrlist(2), so set the last
+        // modified and last access times using utimensat(2)
+        if (lastModifiedTime != null || lastAccessTime != null) {
+            // if not changing both attributes then need existing attributes
+            if (lastModifiedTime == null || lastAccessTime == null) {
                 try {
-                    if (useLutimes)
-                        lutimes(path, accessValue, modValue);
-                    else
-                        futimens(fd, accessValue, modValue);
+                    UnixFileAttributes attrs = UnixFileAttributes.get(path, followLinks);
```

Comment:
We should use openForAttributeAccess if possible. If lastModifiedTime or lastAccessTime is not provided then it ensures that this method will only access one file to avoid racing with concurrent changes on the file system.